Survey/Survey/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift

```
@@ -0,0 +1,40 @@
+//
+//  NetworkAPIProtocol.swift
+//  Survey
+//
+//  Created by Taher on 4/11/22.
+//
+
+import Combine
+import Foundation
+
+protocol NetworkAPIProtocol {
+
+    func performRequest<T: Decodable>(_ configuration: RequestConfiguration, for type: T.Type)
+        -> AnyPublisher<T?, Error>
+}
```

IMO the `AnyPublisher` type contains either value (`T`) and error (`Error`), we don't want to return optional value. Because if for any reason we can't return a non-nil value (invalid API response for eg, ...) we can return an error instead. I'd prefer the return type is `AnyPublisher<T, Error>`. What do you think?

---

updated in this [commit](https://github.com/Shayokh144/ic-survey-swiftui-combine/pull/43/commits/1aab549f658f6030d71ed30af0d2429788c73d4b)